More portability improvements

jamc-code opened the pull request:
Hi! While creating the python script for country -> country code conversion, I ended up finding ways to improve portability.
Mainly, this is done with a `grep` call instead of the `cuts` we had previously. I also added a var called "VPN_PROVIDER" to allow users to clearly set that from the get go, and we can also use it in the `sed` script you mentioned to configure the main script.

Let me know what you think and have a great day!!
    - Julia

jamc-code committed:
catching up to upstream

jamc-code committed:
country code converter started

jamc-code committed:
vpn status finder made portable

jamc-code committed:
embarassing typos

jamc-code committed:
basic untested integration of converter

jamc-code committed:
quick fix for quoting and python integration

jamc-code commented on vpn_module.sh:
```diff
@@ -37,10 +37,11 @@ VPN_RELAY_SET_LOCATION="mullvad relay set location"
-CONNECTED=Connected
-CONNECTING=Connecting
+# Note from Julia: this should cover most VPNs, if it's missing something let me know
+VPN_STATUS="$($VPN_GET_STATUS | grep -Eio 'connected|connecting|disconnected' \
+	| tr '[:upper:]' '[:lower:]')"
+CONNECTED=connected
+CONNECTING=connecting
```
run vpn status command, look for keywords and translate it to lower case. the translation is so vpn providers using {CONNECTED,Connected,connected} will all register the same

jamc-code commented on country_codes.py:
```diff
@@ -49,5 +52,4 @@
 else:
-    print("Not in country dict.")
-    exit(1)
+    exit(2)
```
the different exit codes are so we can handle errors properly in the main script

shervinsahba commented on vpn_module.sh:
```diff
@@ -119,61 +131,63 @@ vpn_location_menu() {
 			-sep "|" -dmenu -i -p "$rofi_menu_name" <<< \
 			" $icon_connect (dis)connect| $icon_fav ${VPN_LOCATIONS[0]}| $icon_fav ${VPN_LOCATIONS[1]}| $icon_fav ${VPN_LOCATIONS[2]}| $icon_fav ${VPN_LOCATIONS[3]}| $icon_fav ${VPN_LOCATIONS[4]}| $icon_fav ${VPN_LOCATIONS[5]}| $icon_fav ${VPN_LOCATIONS[6]}| $icon_fav ${VPN_LOCATIONS[7]}| $icon_country ${VPN_LOCATIONS[8]}| $icon_country ${VPN_LOCATIONS[9]}| $icon_country ${VPN_LOCATIONS[10]}| $icon_countr ...
 
+		# TODO(julia) can these be quoted? i feel like last time it broke something
```
The `VPN_CODES[0]` through `[9]` break if they have quotes for the stock Mullvad setup. Originally, they are passing a country and city code as two arguments to `mullvad relay set location`, so they need to be globbed for that behavior. 

The remaining ones are country codes only, so should be fine in quotes.

The `$rofi_location` may not work with quotes either, for similar reasons, if I remember right. 

Thanks for all these contributions Julia! It's a busy week for me, but I'll test it out on my end soon. These look great.

jamc-code replied:
ahhhhh that's what it was! thanks for the explanation, i wasn't really getting it. also, no problem! i'm happy to help out, it's a super useful module.

jamc-code committed:
add the log spam issue to the README

jamc-code committed:
--no-geoip ready to run when integrated

jamc-code commented:
Hey! I cleaned up the integration of the country code converter and updated the README to reflect that logging issue you found the other week. Hope you're doing well!
\- Julia
